vm/channel.go

```diff
@@ -118,16 +130,26 @@ func builtinChannelInstanceMethods() []*BuiltinMethodObject {
 			// puts(r)
 			// ```
 			//
-			// It takes no argument. TODO: add argument checking.
+			// If you call `close` twice against the same channel, an error is returned.
+			//
+			// It takes no argument.
 			//
 			// @return [Null]
 			Name: "close",
 			Fn: func(receiver Object, sourceLine int) builtinMethodBody {
 				return func(t *Thread, args []Object, blockFrame *normalCallFrame) Object {
+					if len(args) != 0 {
+						return t.vm.InitErrorObject(errors.ArgumentError, sourceLine, "Expect 0 argument. got: %d", len(args))
+					}
 					c := receiver.(*ChannelObject)
 
-					close(c.Chan)
+					if c.ChannelState == CLOSED {
+						return t.vm.InitErrorObject(errors.InternalError, sourceLine, "The channel is already closed. got: %d", len(args))
```

Comment:
I think we should have a dedicated error type for this

Reply:
Got it

Reply:
Updated